src/runtime_src/core/edge/drm/zocl/zocl_xclbin.c

```diff
@@ -197,6 +198,23 @@ zocl_load_pskernel(struct drm_zocl_dev *zdev, struct axlf *axlf)
 	return 0;
 }
 
+static int
+zocl_check_sect(enum axlf_section_kind kind, struct axlf *axlf_full)
```

Comment:
I assume the signature of this function is return the size of the given section. So static size_t or ssize_t is better.

But given it a second thought, you don't even need this function. There is an existing API you can do the same thing in a simpler way because we don't actually care about the size, right?

Check out xrt_xclbin_get_section_num see if you can use it instead of this function.